pkg/aws/aws.go

```diff
@@ -215,6 +223,23 @@ func GetDefaultAMI(ctx context.Context, cfg aws.Config, instanceType string) (st
 	return *result.Images[0].ImageId, nil
 }
 
+func GetAMIRootDevice(ctx context.Context, cfg aws.Config, diskImage string) (string, error) {
```

Comment:
what do you think of falling back to `/dev/sda1` if we encounter an issue querying the images?

Reply:
Can do!  My goal is to make it completely transparent for current users so that makes sense!

I could not find any "guarantee" in the AWS API spec that this field would exist, but I also couldn't find any AMI where it did not exist, so I was kinda stuck figuring out a good assumption :)

Reply:
672ee4a